python/ray/data/block.py

```diff
@@ -374,6 +378,11 @@ def batch_to_block(
                 try:
                     return cls.batch_to_arrow_block(batch)
                 except ArrowConversionError as e:
+                    logger.warning(
+                        f"Failed to convert batch to Arrow due to: {e}; "
+                        f"falling back to Pandas block"
+                    )
```

Comment:
nit: should we move this into the following `if` statement, so that it's only logged when we do the actual conversion to pandas block and not prior to the else case?